### ISSUE TITLE
Getting the i-th sample in dataset

### ISSUE THREAD

**Issue by CLemeunier** [webdataset/webdataset#82]
Hi, 

Is there a way to rapidly access the i-th sample in a dataset? 

What I do for now is the following: 

- creating a dataloader without shuffling and with a batch size of 1: 
```
dataset = wds.Dataset(url).decode().to_tuple("json", "pth")
dataloader = torch.utils.data.DataLoader(dataset, num_workers=0, batch_size=1)
```

- iterate over the dataloader, and if `j==nb_desired_sample`, break the iteration:
```
for j, data in enumerate(dataloader, 0):
    if j == nb_desired_sample:
        labels, inputs = data
        break
```

Is there a faster way to do this? 

Thanks

**Comment by tmbdev** [comment]
You can use `slice`, to select slices of a dataset:

```
dataset = wds.Dataset(url).slice(i, 999999999).decode().to_tuple("json", "pth")
dataloader = torch.utils.data.DataLoader(dataset, num_workers=0, batch_size=1)
```
